benchmark/Streamly/Benchmark/Data/Stream/Common.hs

```diff
@@ -0,0 +1,56 @@
+-- |
+-- Module      : Stream.Common
+-- Copyright   : (c) 2018 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+
+module Stream.Common
+    ( sourceUnfoldr
+    , sourceUnfoldrM
+    , sourceUnfoldrAction
+    , benchIOSink
+    )
+where
+
+import Streamly.Internal.Data.Stream.Type (Stream, fromStreamK)
+import qualified Streamly.Internal.Data.Stream.StreamK.Type as K
+import Control.DeepSeq (NFData)
+import Gauge
+import Prelude hiding (mapM)
+import System.Random (randomRIO)
+
+{-# INLINE sourceUnfoldrM #-}
+sourceUnfoldrM :: (Monad m) => Int -> Int -> Stream m Int
+sourceUnfoldrM count start = fromStreamK $ K.unfoldrMWith K.consM step start
```

Comment:
This should perhaps be `D.unfold....` and `fromStreamD`.

The fusion rule is `fromStreamK . toStreamK == id`
`fromStreamD = fromStreamK . toStreamK`
Serial elimination is of the form `D.someEliminator . toStreamD`
So if the generator is `fromStreamD D.someGenerator`
The entire equation would be `D.someEliminator  toStreamD  fromStreamD D.someGenerator`
Eventually this becomes `D.someEliminator D.someGenerator`. This will likely fuse.
When you use `K` it will definitely not fuse.